### PR TITLE
UP-641 Twig Education Design

### PR DESCRIPTION
**Pull request by IoannisCleary**
**NOTE** Minor improvements added with the design update, main form uses bootstrap now and added a note under the generated graphic as not selecting further down and upwards causes misalignment (possibly due some element being partially selected)

### CODE REVIEW

File: emailsig/views.py [hunk 1]
```diff
@@ -14,8 +14,10 @@ def form_valid(self, form):
         self.request.session['job_title'] = form.cleaned_data['job_title']
         self.request.session['main_contact_number'] = form.cleaned_data['main_contact_number']
         self.request.session['additional_number'] = form.cleaned_data['additional_number']
-        self.request.session['email'] = form.cleaned_data['email']
-        self.request.session['skype'] = form.cleaned_data['skype']
+        self.request.session['address_line_1'] = form.cleaned_data['address_line_1']
+        self.request.session['address_line_2'] = form.cleaned_data['address_line_2']
+        self.request.session['address_line_3'] = form.cleaned_data['address_line_3']
+        self.request.session['website'] = form.cleaned_data['website']
```
Comment (cjwright83) [inline emailsig/views.py]: It would be a lot easier I think to just right this as:

```python
self.request.session.update(form.cleaned_data)
```

**Review by cjwright83 — CHANGES_REQUESTED** [review]

**Review by cjwright83 — APPROVED** [review]

File: emailsig/templates/base.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<head>
+<!-- Latest compiled and minified CSS -->
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
```
Comment (tjaniczek) [inline emailsig/templates/base.html]: As far as I can see you're pulling the whole Boostrap to use a single `.container` class. Highly inefficient

**Review by tjaniczek — DISMISSED** [review]

File: emailsig/templates/base.html (hunk 2 above)
Comment (IoannisCleary) [reply emailsig/templates/base.html]: @tjaniczek  also use the 'well' class :p

Comment (adamoliver) [reply emailsig/templates/base.html]: A lot of email clients won't pull in any external files so I don't think we could do this anyway.

File: emailsig/templates/base.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+<!-- Optional theme -->
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css" integrity="sha384-rHyoN1iRsVXV4nD0JutlnGaslCJuC7uwjduW9SVrLvRYooPp2bWYgmgJQIXwl/Sp" crossorigin="anonymous">
+
+<!-- Latest compiled and minified JavaScript -->
+ <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
```
Comment (adamoliver) [inline emailsig/templates/base.html:10]: Unfortunately you can't include any javascript within html emails.

File: emailsig/templates/email_signature.html [hunk 4]
```diff
@@ -1,22 +1,69 @@
...
+                                    <a href="https://www.instagram.com/twig.world/"><img src="https://res.cloudinary.com/dodge7ws8/image/upload/c_scale,f_png,h_22/v1517315583/email-signature/instagram_logo.svg" alt="Instagram"/></a>
+                                </td>
+                            </tr>
+                        </table>
+                    </td>
+                    <td width="20"></td>
+                    <td>
+                        <table height="120">
+                            <tr style="font-size: 18px; color: #5C5C5C; font-weight: bold; height:60!important; vertical-align: top;">
+                                <td class="name">
```
Comment (adamoliver) [inline emailsig/templates/email_signature.html]: Classes should be avoided on html email as many clients don't use them. Styles should be inline.

Comment (IoannisCleary) [reply emailsig/templates/email_signature.html]: @adamoliver thats fine the class there doesnt add any styling it should be removed

File: emailsig/templates/base.html (hunk 3 above)
Comment (IoannisCleary) [reply emailsig/templates/base.html:10]: @adamoliver javascript is only used on the apps interface

File: emailsig/templates/base.html (hunk 2 above)
Comment (IoannisCleary) [reply emailsig/templates/base.html]: @adamoliver its mainly used for the apps interface

Comment (IoannisCleary) [reply emailsig/templates/base.html]: @adamoliver template works fine in gmail and mac client

File: emailsig/templates/base.html (hunk 3 above)
Comment (adamoliver) [reply emailsig/templates/base.html:10]: Sorry, I thought this was the email template.

File: emailsig/templates/base.html (hunk 2 above)
Comment (adamoliver) [reply emailsig/templates/base.html]: Unfortunately that only accounts for 34%.
https://emailclientmarketshare.com/

**Pull request merged**